Updated GPIO pins for v2 using scematics provided and added klipper config file

eddwatts opened the pull request:
Fixed GPIO pins as they did not match the scematics for the V2 board (matched the pinouts for the V1 board)
Added a Klipper config file

Untested as I do not have the board yet but am looking at it as an option for my printer.

eddwatts committed:
Update README.md

Update to v2 pinouts using schematics provided

eddwatts committed:
Update README.md

updated to GPIO pins for v2 using schematics provided
Added Klipper config

eddwatts committed:
Create PITB.cfg

Klipper config example

eddwatts committed:
Update README.md

eddwatts committed:
Update PITB.cfg

ejhoness committed:
name lowercase and !pit:pin

ejhoness committed:
^!PITB:gpio*

ejhoness committed:
Update PITB.cfg

microstepps:32 and run_current:1.1

ejhoness committed:
Update PITB.cfg

add example 5160

eddwatts committed:
Merge pull request #1 from ejhoness/patch-1

Patch 1

NV4RE commented:
Hi @eddwatts,

Great job on updating the GPIO pins and adding the Klipper config file! I have a few suggestions and observations:

1. **Typo in `printer.cfg`**:
   - The line `endstop_pin: tmc2130_stepper_x:virtual_endstop` should be corrected to `endstop_pin: TMC5160_stepper_x:virtual_endstop`.

2. **Code Block for `printer.cfg`**:
   - It would be helpful to put the `printer.cfg` content into a code block for better readability. You can use [Markdown syntax-highlighting](https://www.markdownguide.org/extended-syntax/#syntax-highlighting) to achieve this.

3. **Sensorless Homing Issue**:
   - I am having trouble getting sensorless homing to work. I'm not sure if there are some jumpers that need to be connected or if there is another configuration step that I might be missing. Any insights or additional documentation on this would be greatly appreciated.

Thanks for your efforts on this update!

eddwatts committed:
Update PITB.cfg

added activity led

eddwatts committed:
Create sensorless_homing.cfg

eddwatts committed:
Update PITB.cfg

eddwatts committed:
Update README.md

Adding firmare update code

eddwatts committed:
Update README.md

eddwatts committed:
Update README.md

added links to klipper config andf sensorless homeing

eddwatts committed:
Update README.md

eddwatts committed:
Update README.md

eddwatts committed:
Add files via upload

eddwatts committed:
Update PITB.cfg

added [display]

eddwatts commented:
As my board has now arrived I have started to imploment the changes, added Firmare flashing procedures including Canboot/Katapult to make updates easier.

Moved the klipper to its own config file and a link to its config rather than showing all the code on the readme.
added [display] and [output_pin activity_led] options and corrected the tmc2130 error

Started work on the sensorless homeing, this should be most of the code but I need to confirm the code is working fully.
The sensorless homeing code is based on the voron guide: https://docs.vorondesign.com/community/howto/clee/sensorless_xy_homing.html
and updated using the sampel code I found for the v1 PITB: https://github.com/kageurufu/PITB/blob/main/firmware/sensorless_homing.cfg

eddwatts committed:
Update README.md

eddwatts committed:
Add files via upload

eddwatts committed:
Create temp

eddwatts committed:
Add files via upload

eddwatts committed:
Delete V2.0/STL/temp

eddwatts committed:
Add files via upload

eddwatts committed:
Update README.md

eddwatts committed:
Update README.md

NV4RE commented:
> As my board has now arrived I have started to imploment the changes, added Firmare flashing procedures including Canboot/Katapult to make updates easier.
> 
> Moved the klipper to its own config file and a link to its config rather than showing all the code on the readme. added [display] and [output_pin activity_led] options and corrected the tmc2130 error
> 
> Started work on the sensorless homeing, this should be most of the code but I need to confirm the code is working fully. The sensorless homeing code is based on the voron guide: https://docs.vorondesign.com/community/howto/clee/sensorless_xy_homing.html and updated using the sampel code I found for the v1 PITB: https://github.com/kageurufu/PITB/blob/main/firmware/sensorless_homing.cfg

oh, you managed to make sensorless homing work! I can't seem to get it to function even after reducing the homing current to 0.3A and setting the driver_SGT to -64.

eddwatts committed:
Update PITB.cfg

eddwatts commented:
@NV4RE 

I had sensorless homeing working when i had tmc2209 drives before adding the PITB, trying to get it back working now.

Im still working on the sensorless homeing, getting lots errors to do with writing to TMC of spi.
Unable to write tmc spi 'stepper_y' register COOLCONF
Unable to write tmc spi 'stepper_x' register GCONF
Unable to write tmc spi 'stepper_x' register TCOOLTHRS

And getting the errors when running:
SET_TMC_FIELD FIELD=SGT STEPPER=stepper_x VALUE=-64 or SET_TMC_FIELD FIELD=SGT STEPPER=stepper_y VALUE=-64
and if i dont get an error then then when running G28 X0 or G28 Y0

I am able to read information from the drivers using DUMP_TMC stepper=stepper_x or DUMP_TMC stepper=stepper_y

Are you suiong 24v/48v or dual 24v system? (im using a dual 24v system here but from what iv read and seen should not be causing these problems)

NV4RE commented:
> @NV4RE
> 
> I had sensorless homeing working when i had tmc2209 drives before adding the PITB, trying to get it back working now.
> 
> Im still working on the sensorless homeing, getting lots errors to do with writing to TMC of spi. Unable to write tmc spi 'stepper_y' register COOLCONF Unable to write tmc spi 'stepper_x' register GCONF Unable to write tmc spi 'stepper_x' register TCOOLTHRS
> 
> And getting the errors when running: SET_TMC_FIELD FIELD=SGT STEPPER=stepper_x VALUE=-64 or SET_TMC_FIELD FIELD=SGT STEPPER=stepper_y VALUE=-64 and if i dont get an error then then when running G28 X0 or G28 Y0
> 
> I am able to read information from the drivers using DUMP_TMC stepper=stepper_x or DUMP_TMC stepper=stepper_y
> 
> Are you suiong 24v/48v or dual 24v system? (im using a dual 24v system here but from what iv read and seen should not be causing these problems)


I'm using 48V and 24V, but Klipper homes without an error message; however, the DIAG pin never triggers.
I am currently using a serial setup instead of CAN because it is kind of unstable for some reason. I plan to switch all devices to a CAN network after everything works fully.

Here is my config:

printer.cfg
```ini
...

[stepper_x]
step_pin:pitb:gpio13
dir_pin:pitb:gpio23
enable_pin:!pitb:gpio22
microsteps: 16
rotation_distance: 40
full_steps_per_rotation: 400
#endstop_pin:^!pitb:gpio16
endstop_pin: tmc5160_stepper_x:virtual_endstop

position_min: -10
position_endstop: 350
position_max: 350  # margin for skew

homing_speed: 80  #Max 100
homing_retract_dist: 0
homing_positive_dir: true

[tmc5160 stepper_x]
cs_pin:pitb:gpio21
spi_software_sclk_pin:pitb:gpio2
spi_software_mosi_pin:pitb:gpio3
spi_software_miso_pin:pitb:gpio4
diag1_pin:!pitb:gpio14
run_current: 1

driver_SGT: -64 # -64 is most sensitive value, 63 is least sensitive
#stealthchop_threshold: 999999
interpolate: False

[stepper_y]
step_pin:pitb:gpio6
dir_pin:pitb:gpio5
enable_pin:!pitb:gpio20
microsteps: 16
rotation_distance: 40
full_steps_per_rotation: 200
#endstop_pin:^!pitb:gpio17
endstop_pin: tmc5160_stepper_y:virtual_endstop

position_min: -10
position_endstop: 360
position_max: 360 # margin for skew

homing_speed: 80   #Max 100
homing_retract_dist: 0
homing_positive_dir: true

[tmc5160 stepper_y]
cs_pin:pitb:gpio1
spi_software_sclk_pin:pitb:gpio2
spi_software_mosi_pin:pitb:gpio3
spi_software_miso_pin:pitb:gpio4
diag1_pin:!pitb:gpio7
run_current: 1
driver_SGT: -64 # -64 is most sensitive value, 63 is least sensitive
#stealthchop_threshold: 999999
interpolate: False

```

sensorless.cfg
```ini
[gcode_macro _HOME_X]
gcode:
    ## Home
    G28 X0
    ## Move away
    G91
    G1 X-10 F2000


[gcode_macro _HOME_Y]
gcode:
    # Home
    G28 Y0
    # Move away so the x-axis is no longer touching the back of the gantry
    G91
    G1 Y-10 F2000

[homing_override]
axes: xyz
gcode:
  {% set home_all = 'X' not in params and 'Y' not in params and 'Z' not in params %}

  {% if home_all or 'X' in params or 'Y' in params%}
    {% set RUN_CURRENT_X = printer.configfile.settings['tmc5160 stepper_x'].run_current|float %}
    {% set RUN_CURRENT_Y = printer.configfile.settings['tmc5160 stepper_y'].run_current|float %}
    {% set RUN_ACCEL = printer.configfile.settings['printer'].max_accel|int %}	
    {% set HOME_CURRENT = 0.3 %}  ## Change this to the vlaue you used when calibrating stallguard. In your printer.cfg.
    SET_TMC_CURRENT STEPPER=stepper_x CURRENT={HOME_CURRENT}
    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={HOME_CURRENT}
    
    UNSAFE_RAISE_TOOL 
    G4 P1000
  {% endif %}

  {% set HOME_ACCEL = 800 %}
  SET_VELOCITY_LIMIT ACCEL={HOME_ACCEL}
  
  {% if home_all or 'X' in params %}
    _HOME_X
    G4 P500
  {% endif %}
  
  {% if home_all or 'Y' in params %}
    _HOME_Y
  {% endif %}

  ##  Set the current back to what's in the printer.cfg file. 
  {% if home_all or 'X' in params or 'Y' in params%}
    SET_TMC_CURRENT STEPPER=stepper_x CURRENT={RUN_CURRENT_X}
    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={RUN_CURRENT_Y}
    G4 P1500
    ## Only uncomment if the code to modify acceleration above was uncommented.
    SET_VELOCITY_LIMIT ACCEL={RUN_ACCEL}
  {% endif %}
  
  {% if home_all or 'Z' in params %}
    G90
    # G1 X175 Y175 F12000
    
    G28 Z
    G1 Z30
  {% endif %}
  
  ```
  
  Additionally, I'm not even sure if the schematic they provided is an up-to-date version or accurate.
  
  The driver_0 toasted 2 of my stepper motors, and I have no idea why, lol.

eddwatts commented:
@NV4RE 

Just been reading up as thought the usb was using gpio0&1 but seems on the pico they are diffrent. so could even make a good usb can bridge?

I cant see any errors in the setup but yet its not working.

Tested the display on i2c0g last night and that worked on so another little function the board can do.
have also tested a BME280 sensor on the i2c0g connector as well, tested fan outputs and temp inputs.
untested so far is the rgb output and endstops.

Also thinking that the awg for the 24v supply is to small when being shared with a ebb36, extruder heater and extruder stepper, this could be cauing power brounouts creating odd errors as well.
48v rail is used to create the 12v rail that is only used on the tmc5160
24v rail is used to create 5v rail
5v rail is used to create the 3.3v rail
but then again the chips there using should eb able to be voltage tolrent to an extent.

next step i think is to improve the 24v supply and see if that changes things as seen in the issues section that rewireing it helped. will update if that helps

eddwatts committed:
Update PITB.cfg

added BME280 i2c sensor option

eddwatts committed:
Update README.md

eddwatts commented:
@NV4RE 
I have now tested a BME280 and a display on the i2c0g port work fine, so thats fans, temps and i2c tested ok
not tested endstops and rgb yet

Iv also tryed a thicker wire for the 24v and confirmed that i have voltage on the MV (48v) and 24v rail.
checked earth points are ok
checekd canbus from mainboard to PITB to ebb all working and talking ok
checked voltage using the 2nd can port to confirm all wires connected ok

starting the firmware shows no errors on screen but in checking klipper logs I spotted:
TMC stepper_x failed to init: Unable to write tmc spi 'stepper_x' register GLOBALSCALER
TMC stepper_y failed to init: Unable to write tmc spi 'stepper_y' register GLOBALSCALER

so something is def not right
have also tryed changed the spi_speed to 3500000 as found posts talking about the speed being to fast
if you come accross anything please let me know as starting to annoy me now lol - and yes still working on it

the driver damaging your steppersm, have you checked the pinout of motors and the driver board? makesure your connecting the correct coils in order? - checked the ones that came in the pack and they seem to be right for my e3d high torque motors.

A bit of help from the designers or makers of the board would probably help here

NV4RE commented:
> @NV4RE I have now tested a BME280 and a display on the i2c0g port work fine, so thats fans, temps and i2c tested ok not tested endstops and rgb yet
> 
> Iv also tryed a thicker wire for the 24v and confirmed that i have voltage on the MV (48v) and 24v rail. checked earth points are ok checekd canbus from mainboard to PITB to ebb all working and talking ok checked voltage using the 2nd can port to confirm all wires connected ok
> 
> starting the firmware shows no errors on screen but in checking klipper logs I spotted: TMC stepper_x failed to init: Unable to write tmc spi 'stepper_x' register GLOBALSCALER TMC stepper_y failed to init: Unable to write tmc spi 'stepper_y' register GLOBALSCALER
> 
> so something is def not right have also tryed changed the spi_speed to 3500000 as found posts talking about the speed being to fast if you come accross anything please let me know as starting to annoy me now lol - and yes still working on it
> 
> the driver damaging your steppersm, have you checked the pinout of motors and the driver board? makesure your connecting the correct coils in order? - checked the ones that came in the pack and they seem to be right for my e3d high torque motors.
> 
> A bit of help from the designers or makers of the board would probably help here

I think I had the same problem with software SPI before but on a different board (BTT Manta M8P). I changed the spi_speed to 2000000, and it fixed the issue.
Note that I did not daisy chain the connector to the EBB36; instead, I crimped new wires directly from the electronics bay.
Also, I have ordered the PITB v1.

eddwatts commented:
@NV4RE
Pitty this spi error is causing such a problem, would have been a good board. I hope the V1 works better for you - was it just the spi fault that made you change?

as a side note, last night my printer tryed an update it dint like and wopuld no longer boot, Had backups of all configs, So 1 full reload later from scratch and loading the configs and updating all boards to there new versions im still working on the spi error.

Just tryed the 2000000 speed with no luck. wish they had sued a hardware spi by swapping the fan0 and the RX pins round. (think this was ene right on the v1 board so why change it????)

think the pcb diagram would help just confiurm pins

eddwatts committed:
Update README.md

eddwatts committed:
Update README.md

eddwatts committed:
Update README.md

eddwatts committed:
Update README.md

NV4RE commented:
> @NV4RE Pitty this spi error is causing such a problem, would have been a good board. I hope the V1 works better for you - was it just the spi fault that made you change?
> 
> as a side note, last night my printer tryed an update it dint like and wopuld no longer boot, Had backups of all configs, So 1 full reload later from scratch and loading the configs and updating all boards to there new versions im still working on the spi error.
> 
> Just tryed the 2000000 speed with no luck. wish they had sued a hardware spi by swapping the fan0 and the RX pins round. (think this was ene right on the v1 board so why change it????)
> 
> think the pcb diagram would help just confiurm pins

No, everything else worked for me, but I can't make the sensorless homing work.

eddwatts committed:
Update README.md

NV4RE commented:
### Update

After switching to a limit switch as the endstop instead of using sensorless homing, I can confirm that this pull request document is accurate.

### Notes for Developers / @FYSETC

1. **Inclusion of Male Terminals**: It would be beneficial to include male terminals in the package. Additionally, providing a terminal for the stepper motor would be a great advantage. I personally find this feature quite useful.

2. **MicroFit Connectors**: Instead of pre-crimping the MicroFit connectors, consider providing 3 or 4 uncrimped connectors. This allows users to crimp them according to their specific needs, especially since the 24V wire included is too thin for pre-crimped connectors.

3. **Documentation Quality**: The current documentation is subpar for a commercial product. It should be improved to reflect the quality and professionalism expected from a product intended for sale, rather than appearing as if it were a charitable effort.

4. **Product Evolution**: The V2 version feels more like a cost-cutting measure rather than an upgrade from V1. This perception could be improved by ensuring that the enhancements in V2 are clearly communicated and genuinely add value over V1.

eddwatts committed:
Create help.txt

eddwatts committed:
Add files via upload

eddwatts committed:
Add files via upload

eddwatts committed:
Create README.md

eddwatts committed:
Add files via upload

NV4RE commented:
Hi, my PITB-V1 arrived and initially worked fine, but I'm getting an error: Unable to write tmc spi 'stepper_x' register GLOBALSCALER. Even after swapping the config from X to Y, the issue persists. I suspect the TMC5160 driver might be faulty. I'm planning to rebuild the printer using BTT parts instead.

eddwatts commented:
Working with support to see if Iv done  anythig  wrong or I have a bad board or there is a fault somewhere.
So far nothing is making the steppers move, even removing the SPI and trying to use the step and dir pins is not working. pitty this looked such a good board.

NV4RE commented:
> Working with support to see if Iv done anythig wrong or I have a bad board or there is a fault somewhere. So far nothing is making the steppers move, even removing the SPI and trying to use the step and dir pins is not working. pitty this looked such a good board.

It seems like there might be a quality control issue. The first board had problems with the Canbus and the DIAC pin wasn't working, while the second board had issues with the TMC5160.

ejhoness commented:
for me at pitbv2 no fan no 5160 
only temp and endstops over canbuS, usbcan and usbserial, 
vendor tell cfg v1 is same to v2  aff. lost much time :(

ejhoness commented:
news pitbv2.cfg but for me no change. and maybe miss ? at 
[5160_y 
cs_pin: PITB:gpio21

Could you tell me if the fans work?

currently working
3 temperature
2 end of stroke sensor

not tested
rgb and extra ios

not working
2 5160
2 fan

eddwatts committed:
Create homing.cfg

eddwatts committed:
Add files via upload

eddwatts committed:
Merge branch 'main' into main

eddwatts commented:
@ejhoness sorry for slow responce, life happens at times.

I can confirm all my fans do work
I now have a replacment board and a 48v psu to complete the upgrade (just need to install) so I hope to provide more updates soon. All parts including the i2c disdplay and sensors were tested on my 1st board  but I never got the motors working on that board.

Thought on the fans, do you have the jumper set up for the fan voltage? if its missing then now power is being supplyed to the fan.

what was the problem you were vaing with the cs_pin?

eddwatts committed:
Update PITB.cfg

ejhoness commented:
Thanks, my board is coming back, sad because 48v source and big motors, for me with new taxes I won't buy other parts any time soon. From what I read this is the solution, include spi pins from pitb in klipper to work.  

              Got the spi bus working: https://github.com/Klipper3d/klipper/pull/6683

_Originally posted by @KoiosLabs in https://github.com/FYSETC/FYSETC-PITB/issues/2#issuecomment-2341296930_

LadPack commented:
Can anyone confirm whether or not any pins need to be jumpered for sensorless homing to work? I have the steppers working over spi via the above fix, but for the life of me I cannot figure out how to get sensorless to work.  Either it doesn't move at all between -64 and 63, or it crashes into the frame

eddwatts committed:
Update PITB.cfg

Updated with info for hardware SPI from https://github.com/Klipper3d/klipper/pull/6683

eddwatts committed:
Delete V2.0/Klipper_Config directory

eddwatts committed:
Delete V2.0/support directory

eddwatts committed:
Update README.md

update to Katapult replacing CANboot
Useing KCONFIG_CONFIG=config.pitb to create its own config away from MCU Firmware

eddwatts committed:
Update README.md

eddwatts commented:
Updayed config for SPI pins now this fix has been done: https://github.com/Klipper3d/klipper/pull/6683
Will test as soon as I can, do wonder if this was the fault all along

eddwatts committed:
Update README.md

added hardware SPI bus infor for TMC Drivers

eddwatts committed:
Update README.md
